extensions/positron-r/src/commands.ts

```diff
@@ -143,24 +143,12 @@ export async function registerCommands(context: vscode.ExtensionContext) {
 			await vscode.commands.executeCommand('workbench.action.languageRuntime.select', 'r');
 		}),
 
-		// Command used to source the current file
+		// Commands used to source the current file
 		vscode.commands.registerCommand('r.sourceCurrentFile', async () => {
-			try {
-				const filePath = await getEditorFilePathForCommand();
-				// In the future, we may want to shorten the path by making it
-				// relative to the current working directory.
-				if (filePath) {
-					const command = `source(${JSON.stringify(filePath)}, echo = TRUE)`;
-					positron.runtime.executeCode('r', command, false);
-				}
-			} catch (e) {
-				// This is not a valid file path, which isn't an error; it just
-				// means the active editor has something loaded into it that
-				// isn't a file on disk.  In Positron, there is currently a bug
-				// which causes the REPL to act like an active editor. See:
-				//
-				// https://github.com/posit-dev/positron/issues/780
-			}
+			sourceCurrentFile(false);
+		}),
+		vscode.commands.registerCommand('r.sourceCurrentFileWithEcho', async () => {
```

Comment:
To avoid increasing our commands list, how about adding an optional parameter to `r.sourceCurrentFile`? Then users can add an `"args"` field to their keybindings to pass the argument if needed.

Reply:
I did try this originally, but AFAICT this would keep us from offering both true/false as editor actions and with different keybindings in the RStudio Keymap; I don't believe an extension can offer two editor actions that are the same command with different args and different display names, like this:

![Screenshot 2024-11-04 at 10 26 58 AM](https://github.com/user-attachments/assets/34d7c498-07e8-4230-b929-c81a4be980c3)